pkgs/by-name/et/ethercat/kernel.nix

```diff
@@ -0,0 +1,199 @@
+{
+  autoreconfHook,
+  pkg-config,
+  stdenv,
+  pkgs,
+  lib,
+  kernel,
+}:
+let
+  deviceKernelCompatabilities = {
+    "6.6" = [ "igc" ];
+    "6.4" = [
+      "e100"
+      "igc"
+    ];
+    "6.1" = [
+      "8139too"
+      "bcmgenet"
+      "e100"
+      # "e1000" # Is broken
+      "e1000e"
+      "igb"
+      # "igc" # Is broken
+      "r8169"
+    ];
+    "5.15" = [
+      "8139too"
+      "e100"
+      "e1000"
+      "e1000e"
+      "igb"
+      "r8169"
+    ];
+    "5.14" = [
+      "8139too"
+      "bcmgenet"
+      "e100"
+      "e1000"
+      "e1000e"
+      "igb"
+      "igc"
+      "r8169"
+    ];
+    "5.10" = [
+      "8139too"
+      "bcmgenet"
+      "e100"
+      "e1000"
+      "e1000e"
+      "igb"
+      "r8169"
+    ];
+    "5.4" = [
+      "e100"
+      "e1000e"
+    ];
+    "4.19" = [ "igb" ];
+    "4.4" = [
+      "8139too"
+      "e100"
+      "e1000"
+      "e1000e"
+      "igb"
+      "r8169"
+    ];
+    "3.18" = [ "igb" ];
+    "3.16" = [
+      "8139too"
+      "e100"
+      "e1000"
+      "e1000e"
+      "r8169"
+    ];
+    "3.14" = [
+      "8139too"
+      "e100"
+      "e1000"
+      "e1000e"
+      "r8169"
+    ];
+    "3.12" = [
+      "8139too"
+      "e100"
+      "e1000"
+      "e1000e"
+      "r8169"
+    ];
+    "3.10" = [
+      "8139too"
+      "e100"
+      "e1000"
+      "e1000e"
+      "r8169"
+    ];
+    "3.8" = [
+      "8139too"
+      "e100"
+      "e1000"
+      "e1000e"
+      "r8169"
+    ];
+    "3.6" = [
+      "8139too"
+      "e100"
+      "e1000"
+      "e1000e"
+      "r8169"
+    ];
+    "3.4" = [
+      "8139too"
+      "e100"
+      "e1000"
+      "e1000e"
+      "r8169"
+    ];
+    "3.2" = [
+      "8139too"
+      "e1000e"
+      "r8169"
+    ];
+    "3.0" = [
+      "8139too"
+      "e100"
+      "e1000"
+    ];
+  };
+
+  kernelVersionMajorMinor = builtins.concatStringsSep "." (
+    lib.lists.take 2 (builtins.splitVersion kernel.version)
+  );
+
+  supportedDrivers = deviceKernelCompatabilities.${kernelVersionMajorMinor};
+in
+stdenv.mkDerivation (
+  finalAttrs:
+  (
+    with pkgs.ethercat;
```

Comment:
We do not want to use with's over such a large scope

Reply:
Resolved in 14a1b28727fac1532a2b21269c0f5f0d317e4405